pkg/jobs/update.go

```diff
@@ -135,31 +155,24 @@ func (u Updater) update(ctx context.Context, updateFn UpdateFn) (retErr error) {
 		log.VInfof(ctx, 1, "job %d: update called with no session ID", j.ID())
 	}
 
-	md := JobMetadata{
-		ID:       j.ID(),
-		Status:   status,
-		Payload:  payload,
-		Progress: progress,
-	}
-
-	offset := 0
-	if j.session != nil {
-		offset = 1
-	}
-	var lastRun *tree.DTimestamp
-	var ok bool
-	lastRun, ok = row[3+offset].(*tree.DTimestamp)
+	lastRun, ok := row[4].(*tree.DTimestamp)
 	if !ok {
 		return errors.AssertionFailedf("expected timestamp last_run, but got %T", lastRun)
 	}
-	var numRuns *tree.DInt
-	numRuns, ok = row[4+offset].(*tree.DInt)
+	numRuns, ok := row[5].(*tree.DInt)
```

Comment:
to confirm its okay to drop the offsetting logic because this will just be NULL if the job doesn't have a claim_session_id?